pkgs/applications/editors/gnome-builder/default.nix

```diff
@@ -0,0 +1,83 @@
+{ stdenv, fetchurl, gnome3, gobjectIntrospection, meson, ninja, pkgconfig
+, appstream-glib, desktop-file-utils, python3, python3Packages, wrapGAppsHook
+, gspell, gtk3, gtksourceview3, json-glib, jsonrpc-glib, libdazzle, libxml2, pcre
+, sysprof, template-glib, vala, webkitgtk
+}:
+let
+  version = "3.28.4";
+  pname = "gnome-builder";
+in stdenv.mkDerivation {
+  name = "${pname}-${version}";
+
+  src = fetchurl {
+    url = "mirror://gnome/sources/${pname}/${gnome3.versionBranch version}/${pname}-${version}.tar.xz";
+    sha256 = "0ibb74jlyrl5f6rj1b74196zfg2qaf870lxgi76qzpkgwq0iya05";
+  };
+
+  buildInputs = [
+    python3Packages.wrapPython
+    wrapGAppsHook
+    gobjectIntrospection
+    meson
+    ninja
+    pkgconfig
+
+    appstream-glib
+    desktop-file-utils
+
+    gnome3.devhelp
+    gnome3.libgit2-glib
+    gnome3.libpeas
+    gnome3.vte
+    gspell
+    gtk3
+    gtksourceview3
+    json-glib
+    jsonrpc-glib
+    libdazzle
+    libxml2
+    pcre
+    sysprof
+    template-glib
+    vala
+    webkitgtk
+  ];
+
+  prePatch = ''
+    sed -i 's@/usr/bin/env python3@${python3}/bin/python3@g' build-aux/meson/post_install.py
```

Comment:
We have `patchShebangs` function for this, see https://github.com/NixOS/nixpkgs/pull/33270